### PR TITLE
add rule prefer-intl-name

### PR DESCRIPTION
**Pull request by andrewst**
@dkrutskikh please look

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,3 +93,4 @@ A plugin for the Dart `analyzer` library [package](https://pub.dev/packages/dart
 * [no-empty-block](https://github.com/wrike/dart-code-metrics/blob/master/doc/rules/no_empty_block.md)
 * [no-magic-number](https://github.com/wrike/dart-code-metrics/blob/master/doc/rules/no_magic_number.md)
 * [prefer-trailing-comma-for-collection](https://github.com/wrike/dart-code-metrics/blob/master/doc/rules/prefer_trailing_comma_for_collection.md)
+* [prefer-intl-name](https://github.com/wrike/dart-code-metrics/blob/master/doc/rules/prefer_intl_name.md)
```
Comment (dkrutskikh) [inline README.md:96]: @andrewst please sort rules

File: doc/rules/prefer_intl_name.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+## Description
+Prefer used pattern `${ClassName}_${ClassMemberName}` for `name` argument in `Intl.message()`, `Intl.plural()`, `Intl.gender()`, `Intl.select()` methods.
+
+### Example
+Bad:
+```dart
+import 'package:intl/intl.dart';
+
+class SomeButtonI18n {
```
Comment (dkrutskikh) [inline doc/rules/prefer_intl_name.md:14]: @andrewst add examples with static functions (like in this https://github.com/dart-lang/intl#messages)

File: lib/src/rules_factory.dart [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import 'package:dart_code_metrics/src/rules/prefer_intl_name.dart';
```
Comment (dkrutskikh) [inline lib/src/rules_factory.dart:1]: @andrewst please use relative import

File: doc/rules/prefer_intl_name.md (hunk 2 above)
Comment (andrewst) [reply doc/rules/prefer_intl_name.md:14]: fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 # Unreleased
 - Drop dependency on resource
-- Add experimental static code diagnostic binary-expression-operand-order
+- Add experimental static code diagnostic binary-expression-operand-order, prefer-intl-name
```
[the remark this answers is not in the document]
Comment (andrewst) [reply CHANGELOG.md]: fixed

File: README.md (hunk 1 above)
Comment (andrewst) [reply README.md:95]: fixed

File: lib/src/rules_factory.dart (hunk 3 above)
Comment (andrewst) [reply lib/src/rules_factory.dart]: fixed

File: doc/rules/prefer_intl_name.md (hunk 2 above)
Comment (andrewst) [reply doc/rules/prefer_intl_name.md:14]: fixed

**Pull request merged**